erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -323,9 +323,13 @@ def update_status_and_detail(source, target, parent):
 		target.maintenance_schedule = source.name
 		target.maintenance_schedule_detail = s_id
 
-	def update_sales(source, target, parent):
+	def update_sales_and_serial(source, target, parent):
 		sales_person = frappe.db.get_value('Maintenance Schedule Detail', s_id, 'sales_person')
 		target.service_person = sales_person
+		serial_nos = get_serial_nos(target.serial_no)
+		if len(serial_nos) == 1:
+			target.serial_no = serial_nos[0]
+			return
 		target.serial_no = ''
```

Comment:
```suggestion
		else:
			target.serial_no = ''
```

This clarifies the intent better IMO.